server/py/services/api/launcher.py

```diff
@@ -240,17 +244,25 @@ def _pre_run_node_selector_enrichement(self, runtime, run):
             run.spec.node_selector = resolved_node_selectors
         return run
 
-    def _pre_run_image_pull_secret_enrichment(self, run):
+    def _pre_run_image_pull_secret_enrichment(
+        self,
+        runtime: "mlrun.runtimes.base.BaseRuntime",
+        run: mlrun.run.RunObject,
+    ):
         """
-        Enrich the run object with the project's image pull secret.
+        Enrich both the run object and runtime with the project's image pull secret.
```

Comment:
The reason we would not want to enrich runtime with image pull secret is that it is subjected to its environment - which means, if we move (export/import) the runtime between envs (imagine having staging + prod envs) then the configuration would persist.
what we do is enriching the run (and not the runtime) configuration.

if there is an issue with specific runtime (e.g nuclio) then we do the enrichment specifically for that runtime